service/matching/tasklist/task_list_manager.go

```diff
@@ -227,6 +227,7 @@ func (c *taskListManagerImpl) Stop() {
 	c.liveness.Stop()
 	c.taskWriter.Stop()
 	c.taskReader.Stop()
+	c.matcher.DisconnectBlockedPollers()
```

Comment:
nice, I was wondering about this

Reply:
Yeah. I also wondered how it can be the clients are not disconnected. Turns out, we never disconnect of matcher unless long-polling timeout is exhausted. So, the connected pollers are "holding" pointer to the `Matcher` object even though `*taskListManager` is gone.

Reply:
yeah, I'm near certain I've seen this locally repro'd and was always unsure what to do with it